docs/design/generics/terminology.md

```diff
@@ -637,60 +637,54 @@ conditions on the type argument. For example: `Array(T)` might implement
 
 ## Interface type parameters and associated types
 
-Imagine an interface defining a container. Different containers will contain
-different types of values, and the container API will have to refer to that
-"element type" when defining the signature of methods like "insert" or "find".
-If that element type is a parameter (input) to the interface type, we say it is
-an _interface type parameter_; if it is an output, we say it is an _associated
-type_. An associated type is a kind of [associated entity](#associated-entity).
-
-Interface type parameter example:
-
-```
-interface StackTP(ElementType:! Type)
-  fn Push[addr me: Self*](value: ElementType);
-  fn Pop[addr me: Self*]() -> ElementType;
-}
-```
-
-Associated type example:
+_Interface type parameters_ and _associated types_ are both ways of allowing the
+types in the signatures of function in an interface to vary. For example,
+different [stacks](<https://en.wikipedia.org/wiki/Stack_(abstract_data_type)>)
+will have different element types. That element type would be used as the
+parameter type of the `Push` function and the return type of the `Pop` function.
+As
+[in Rust](https://rust-lang.github.io/rfcs/0195-associated-items.html#clearer-trait-matching):
```

Comment:
This use of "as in" confuses me. I can read this as:

- Rust does the following: (but it doesn't seem to)
- Carbon uses Rust's terminology: (but I don't see "interface type parameter" mentioned)
- Carbon implements this the same way Rust does: (but I assume it doesn't?)

Are you just trying to say that where Rust calls these things "inputs" and "outputs", we instead call them "interface type parameters" and "associated types"? I'm not clear that the Rust comparison is helping here...

Reply:
Is this better? https://github.com/carbon-language/carbon-lang/pull/1972/commits/57df22a8fea4db474f10276725cc310447a29859